### ISSUE TITLE
NET-1904 S3060 secondary locations improvement.

### ISSUE THREAD

**Issue by csaba-sagi-sonarsource** [SonarSource/sonar-dotnet#4634]
Improve locations for S3060  so that only type check parts of the pattern are reported as secondary locations.
For example:
```
if (this is Program and { MyProperty: 2 })
//  ^^^^    ^^^^^^^
```

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [NET-1904](https://sonarsource.atlassian.net/browse/NET-1904)

[NET-1904]: https://sonarsource.atlassian.net/browse/NET-1904?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
